spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/hazelcast/HazelcastAutoConfiguration.java

```diff
@@ -45,6 +46,7 @@
 @ConditionalOnClass(HazelcastInstance.class)
 @ConditionalOnMissingBean(HazelcastInstance.class)
 @EnableConfigurationProperties(HazelcastProperties.class)
+@AutoConfigureAfter(HazelcastClientAutoConfiguration.class)
```

Comment:
Rather than having two distinct auto-configuration, I'd like that we merge those in the same `HazelcastAutoConfiguration` with a `HazelcastClientConfiguration` and a `HazelcastServerConfiguration`. 

It would be nice if we could have a `spring.hazelcast.mode` enum (`HazelcastMode`) that could either be `CLIENT` or `SERVER`. The default value would be to detect what would work in practice at runtime.